app/features/worksheet_generator/tools.py

```diff
@@ -0,0 +1,292 @@
+            credentials, _ = google.auth.default()
+            self.client = VertexAIEmbeddings(
+                model=model_name,
+                project=project,
+                location=location,
+                credentials=credentials
+            )
+            print("VertexAIEmbeddings client initialized successfully")
+        except Exception as e:
+            print(f"Error initializing VertexAIEmbeddings client: {e}")
+        
+    def embed_query(self, query):
+        print(f"Embedding query: {query}")
+        try:
+            vectors = self.client.embed_query(query)
+            print(f"Embeddings retrieved: {vectors}")
+            return vectors
+        except Exception as e:
+            print(f"Error embedding query: {e}")
+            return None
+    
+    def embed_documents(self, documents):
+        print(f"Embedding documents: {documents}")
+        try:
+            return self.client.embed_documents(documents)
+        except Exception as e:
+            print(f"Error embedding documents: {e}")
+            return None
+
+class ChromaCollectionCreator:
+    def __init__(self, processor, embed_model):
+        self.processor = processor
+        self.embed_model = embed_model
+        self.db = None
+    
+    def create_chroma_collection(self):
+        if len(self.processor.pages) == 0:
+            st.error("No documents found!", icon="🚨")
+            return
+
+        text_splitter = CharacterTextSplitter(
+            separator="\n",
+            chunk_size=1000,
+            chunk_overlap=200
+        )
+        
+        texts = text_splitter.split_documents(self.processor.pages)
+
+        if texts:
+            st.success(f"Successfully split pages into {len(texts)} text chunks!", icon="✅")
+        else:
+            st.error("Failed to split documents into chunks!", icon="🚨")
+            return
+
+        try:
+            self.db = Chroma.from_documents(documents=texts, embedding=self.embed_model)
+            st.success("Successfully created Chroma Collection!", icon="✅")
+        except Exception as e:
+            st.error(f"Failed to create Chroma Collection! Error: {e}", icon="🚨")
+    
+    def query_chroma_collection(self, query):
+        if self.db:
+            docs = self.db.similarity_search_with_relevance_scores(query)
+            if docs:
+                return docs[0]
+            else:
+                st.error("No matching documents found!", icon="🚨")
+        else:
+            st.error("Chroma Collection has not been created!", icon="🚨")
+
+class QuizGenerator:
+    def __init__(self, topic=None, num_questions=1, vectorstore=None):
+        self.topic = topic if topic else "General Knowledge"
+        if num_questions > 10:
+            raise ValueError("Number of questions cannot exceed 10.")
+        self.num_questions = num_questions
+        self.vectorstore = vectorstore
+        self.llm = None
+        self.question_bank = []
+        self.system_template = """
```

Comment:
You can try using a prompt in a separated file for better maintainability and keeping DRY.